api/src/middlewares/mongodb-sanitizer.ts

```diff
@@ -0,0 +1,37 @@
+import { NextFunction, Request, Response } from 'express';
+import * as _ from 'lodash';
+import { StatusCodes } from 'http-status-codes';
+
+const BAD_CHAR = '$';
+
+export const mongodbSanitizer = (req: Request, res: Response, next: NextFunction) => {
+	const body = req.body;
+
+	const val = hasBadCharacter(body, BAD_CHAR);
+
+	if (val) {
+		return res.status(StatusCodes.BAD_REQUEST).send({ error: `${BAD_CHAR} is not allowed as key prefix.` });
+	}
+
+	next();
+};
+
+const hasBadCharacter = (data: any, character: string): boolean => {
+	if (_.isObject(data)) {
+		const keys = Object.keys(data);
+		const foundBadChar = keys.map((k) => k.startsWith(character)).some((s) => s);
```

Comment:
You can replace keys.map().some() with 
- keys.find(k => k.startsWith(character))
- keys.filter(k => k.startsWith(character)).length
Just check if is better: i feel that is more expressive than map/some